server/src/domain/asset/asset.service.ts

```diff
@@ -420,11 +427,54 @@ export class AssetService {
       await this.assetRepository.updateAll([options.stackParentId], { stackParentId: null });
     }
 
+    if (creationDate !== undefined || latitude !== undefined || longitude !== undefined) {
+      this.validateGpsCoordinates(latitude, longitude);
+      for (const id of ids) {
+        await this.updateExifData(id, { CreationDate: creationDate, GPSLatitude: latitude, GPSLongitude: longitude });
+      }
+    }
+
     await this.jobRepository.queue({ name: JobName.SEARCH_INDEX_ASSET, data: { ids } });
     await this.assetRepository.updateAll(ids, options);
     this.communicationRepository.send(CommunicationEvent.ASSET_UPDATE, authUser.id, ids);
   }
 
+  validateGpsCoordinates(latitude: number | undefined, longitude: number | undefined): void {
+    if ((latitude !== undefined || longitude !== undefined) && !(latitude !== undefined && longitude !== undefined)) {
+      throw new BadRequestException('Latitude and longitude must be provided together');
+    }
+  }
+
+  async updateExifData(id: string, exifModify: Partial<Tags>): Promise<void> {
+    const { pathMetadata, newExifData } = await this.getPathAndExif(id, exifModify);
+    await this.assetRepository.writeExif(pathMetadata, newExifData);
+    await this.assetRepository.save({ id, sidecarPath: pathMetadata });
+    await this.jobRepository.queue({ name: JobName.METADATA_EXTRACTION, data: { id: id } });
+  }
+
+  async getPathAndExif(id: string, exifModify: Partial<Tags>): Promise<{ pathMetadata: string; newExifData: Tags }> {
+    const asset = await this.assetRepository.getById(id);
+    if (!asset) {
+      throw new BadRequestException('Asset not found');
+    }
+
+    let pathMetadata: string;
+
+    if (!asset.sidecarPath || !fs.existsSync(asset.sidecarPath)) {
```

Comment:
I don't think we should be checking for this, it should not exist unless it is referenced in the database. Also, we should be checking it synchronously if we do want to double check. Basically, we don't want to duplicate the "discover sidecar" job here.

Reply:
Done